src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -235,7 +235,7 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
                     if ($parameter->isDefaultValueAvailable()) {
                         $value = $parameter->getDefaultValue();
                     } elseif (!$parameter->allowsNull()) {
-                        throw new AutowiringFailedException($this->currentId, $failureMessage);
+                        throw new AutowiringFailedException($this->currentId, $failureMessage());
```

Comment:
this should be reverted, passing a closure is on purpose, see AutowiringFailedException

Reply:
Hmm, indeed, the fix was wrong, didn't check the trick in `AutowiringFailedException`

The original issue is in PHPUnit 8
```
Symfony\Component\DependencyInjection\Tests\Compiler\AutowirePassTest::testNotWireableCalls with data set #1 ('setDifferentNamespace', 'Cannot autowire service "foo"...xists.')                                                                         
TypeError: strpos() expects parameter 1 to be string, object given
```

Since PHPUnit 8, the class `\PHPUnit\Framework\Constraint\ExceptionMessage` is set to `declare(strict_types=1)` which fail to execute `return \strpos($other->getMessage(), $this->expectedMessage) !== false;`

Reply:
https://github.com/sebastianbergmann/phpunit/pull/3787

Reply:
use try/catch workarround in tests